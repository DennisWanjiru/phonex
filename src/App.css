.page {
  padding: 20px 40px;
  display: grid;
  height: 98vh;
  grid-template-columns: 20% 1fr;
  grid-gap: 20px;
}

.card {
  background: #fff;
  border: solid #ccc 1px;
}

form {
  margin-top: 20px;
  display: flex;
  flex-direction: column;
}

input {
  border: solid 1px #000;
  padding: 10px;
  margin-bottom: 20px;
  outline: none;
  font-size: 13px;
}

button {
  padding: 10px 20px;
  color: #000;
  border: solid 1px #000;
  cursor: pointer;
  font-size: 12px;
  border-radius: 3%;
  background: linear-gradient(to right, #000 50%, white 50%);
  background-size: 200% 100%;
  background-position: right bottom;
  transition: all .5s ease-out;
  outline: none;
}

button:hover {
  background-position: left bottom;
  color: #fff;
}

.card:first-child {
  padding: 20px 30px;
  display: grid;
  grid-template-rows: 1fr 20px;
  grid-gap: 10px;
}

.card:first-child > p {
  color: #888;
  font-size: 12px;
}

.card:last-child {
  overflow-y: scroll;
}

.row {
  padding: 13px 20px;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.row:not(:last-child) {
  border-bottom: solid 1px #ccc;
}

.row:not(:first-child):hover {
  background: whitesmoke;
  cursor: pointer;
}

.error {
  color: red;
  font-size: 12px;
  margin-top: -20px;
  margin-bottom: 10px;
}

.error-input {
  border-color: red;
  color: red;
}

.form-input {
  display: flex;
  flex-direction: column;
}

.clickable {
  cursor: pointer;
}