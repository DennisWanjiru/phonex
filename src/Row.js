import React from 'react'

export default function Row({ data: { index, phone }, onClick }) {
  return (
    <div className="row">
      <div>{index}</div>
      <div className="clickable" onClick={onClick}>{phone}</div>
    </div>
  )
}
