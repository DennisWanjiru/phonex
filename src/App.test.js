import { shallow } from 'enzyme';
import React from 'react';
import App from './App';
import Row from './Row'

describe('<App />', () => {
  it('renders  <App /> components', () => {
    const snap = shallow(<App />);
    expect(snap).toMatchSnapshot();
  });
});

describe('<Row />', () => {
  it('renders  <Row /> components', () => {
    const snap = shallow(<Row className data={{ index: "Index", phone: 'Phone Numbers' }} onClick={() => {}} />);
    expect(snap).toMatchSnapshot();
  });
});