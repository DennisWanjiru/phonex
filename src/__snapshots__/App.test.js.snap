// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders  <App /> components 1`] = `ShallowWrapper {}`;

exports[`<Row /> renders  <Row /> components 1`] = `ShallowWrapper {}`;
